### PR TITLE
fix load image memory leak

### PR DESCRIPTION
**Pull request by novain1102**
https://github.com/cocos/cocos-engine/issues/11221

### Changelog

* fix load image memory leak

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check // Manual trigger with `@cocos-robot run test cases` afterward

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/core/animation/SkeletalAnimationUtils.cpp [hunk 1]
```diff
@@ -106,7 +102,6 @@ void deleteTransform(Node *node) {
     while (transform != nullptr) {
         iter = pool.find(transform->node->getUuid());
         if (iter != pool.end()) {
-            //            delete iter->second;
```
Comment (novain1102) [inline native/cocos/core/animation/SkeletalAnimationUtils.cpp]: directly delete will crash

File: native/cocos/core/animation/SkeletalAnimationUtils.h [hunk 2]
```diff
@@ -25,18 +25,21 @@
...
+#include "cocos/base/RefCounted.h"
+#include "cocos/base/Ptr.h"
 namespace cc {
 
 class Node;
 
-struct IJointTransform {
-    Node *node{nullptr};
+struct IJointTransform : public RefCounted {
+    IntrusivePtr<Node> node;
```
Comment (dumganhar) [inline native/cocos/core/animation/SkeletalAnimationUtils.h]: Does IJointTransform need to own node ? Will node be released?

File: cocos/core/assets/image-asset.jsb.ts [hunk 3]
```diff
@@ -130,6 +130,18 @@ imageAssetProto.reset = function (data: ImageSource) {
...
+imageAssetProto.destroy = function () {
+    if(this.data && this.data instanceof HTMLImageElement) {
+        this.data.src = '';
+        this._setRawAsset('');
+        this.data.destroy();
+    } else if (isImageBitmap(this.data)) {
+        this.data.close && this.data.close();
+    }
+    return superDestroy.call(this);
+};
```
Comment (dumganhar) [inline cocos/core/assets/image-asset.jsb.ts:143]: Good catch.

File: native/cocos/core/animation/SkeletalAnimationUtils.h (hunk 2 above)
Comment (novain1102) [reply native/cocos/core/animation/SkeletalAnimationUtils.h]: No need for currently. I added it for unity style. It's better to use raw pointer `node`?

Comment (dumganhar) [reply native/cocos/core/animation/SkeletalAnimationUtils.h]: IJointTransform is put into a global `vector`, it's very dangerous to hold the Node in a global container.
It's better to keep weak reference for `node`.

Comment (novain1102) [reply native/cocos/core/animation/SkeletalAnimationUtils.h]: got it

File: native/cocos/core/animation/SkeletalAnimationUtils.h [hunk 4]
```diff
@@ -25,18 +25,21 @@
...
 
-struct IJointTransform {
-    Node *node{nullptr};
+struct IJointTransform : public RefCounted {
+    Node* node{nullptr};
     Mat4 local;
     Mat4 world;
     int stamp{-1};
-    struct IJointTransform *parent{nullptr};
+    IntrusivePtr<IJointTransform> parent;
```
Comment (minggo) [inline native/cocos/core/animation/SkeletalAnimationUtils.h]: Will parent also hold a reference of its child?

Comment (novain1102) [reply native/cocos/core/animation/SkeletalAnimationUtils.h]: @dumganhar I'm not sure, should use IntrusivePtr for parent or just raw ptr?

Comment (dumganhar) [reply native/cocos/core/animation/SkeletalAnimationUtils.h]: The ownership here is complicated, we need to check the original ts code.

Comment (dumganhar) [reply native/cocos/core/animation/SkeletalAnimationUtils.h]: @mmyduckx Please separate this PR into two.
The `imageAssetProto.destroy` one could be merged first.

Comment (novain1102) [reply native/cocos/core/animation/SkeletalAnimationUtils.h]: OK

**Review by dumganhar — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**